src/constants/dom-event-types.ts

```diff
@@ -0,0 +1,935 @@
+export type EventInterface =
+'AnimationEvent' |
+'AudioProcessingEvent' |
+'BeforeInputEvent' |
+'BeforeUnloadEvent' |
+'BlobEvent' |
+'CSSFontFaceLoadEvent' |
+'ClipboardEvent' |
+'CloseEvent' |
+'CompositionEvent' |
+'CustomEvent' |
+'DOMTransactionEvent' |
+'DeviceLightEvent' |
+'DeviceMotionEvent' |
+'DeviceOrientationEvent' |
+'DeviceProximityEvent' |
+'DragEvent' |
+'EditingBeforeInputEvent' |
+'ErrorEvent' |
+'Event' |
+'FetchEvent' |
+'FocusEvent' |
+'GamepadEvent' |
+'HashChangeEvent' |
+'IDBVersionChangeEvent' |
+'InputEvent' |
+'KeyboardEvent' |
+'MediaStreamEvent' |
+'MessageEvent' |
+'MouseEvent' |
+'MutationEvent' |
+'OfflineAudioCompletionEvent' |
+'OverconstrainedError' |
+'PageTransitionEvent' |
+'PaymentRequestUpdateEvent' |
+'PointerEvent' |
+'PopStateEvent' |
+'ProgressEvent' |
+'RTCDataChannelEvent' |
+'RTCIdentityErrorEvent' |
+'RTCIdentityEvent' |
+'RTCPeerConnectionIceEvent' |
+'RelatedEvent' |
+'SVGEvent' |
+'SVGZoomEvent' |
+'SensorEvent' |
+'StorageEvent' |
+'TimeEvent' |
+'TouchEvent' |
+'TrackEvent' |
+'TransitionEvent' |
+'UIEvent' |
+'UserProximityEvent' |
+'WebGLContextEvent' |
+'WheelEvent'
+
+
+export interface DomEvent {
+  eventInterface: EventInterface | string
+  bubbles: boolean
+  cancelable: boolean
+}
+
+export type DomEventName = keyof typeof domEvents
+
+const domEvents = {
+  abort: {
+    eventInterface: 'Event',
+    bubbles: false,
+    cancelable: false
+  },
+  afterprint: {
+    eventInterface: 'Event',
+    bubbles: false,
+    cancelable: false
+  },
+  animationend: {
+    eventInterface: 'AnimationEvent',
+    bubbles: true,
+    cancelable: false
+  },
+  animationiteration: {
+    eventInterface: 'AnimationEvent',
+    bubbles: true,
+    cancelable: false
+  },
+  animationstart: {
+    eventInterface: 'AnimationEvent',
+    bubbles: true,
+    cancelable: false
+  },
+  appinstalled: {
+    eventInterface: 'Event',
+    bubbles: false,
+    cancelable: false
+  },
+  /**
+   * @deprecated
+   */
```

Comment:
There's probably more I just added the missing properties and saw that it was depricated.